File: app/src/main/java/co/tpcreative/supersafe/common/util/ThemeUtil.java
package co.tpcreative.supersafe.common.util;
import java.util.ArrayList;
import co.tpcreative.supersafe.R;
import co.tpcreative.supersafe.model.Theme;

/**
 * Created by Pankaj on 12-11-2017.
 */

public class ThemeUtil {



    public static final int THEME_PRIMARY = 0;
    public static final int THEME_PINK = 1;
    public static final int THEME_PURPLE = 2;
    public static final int THEME_DEEPPURPLE = 3;
    public static final int THEME_CYAN = 4;
    public static final int THEME_TEAL = 5;
    public static final int THEME_LIGHTGREEN = 6;
    public static final int THEME_LIME = 7;
    public static final int THEME_AMBER = 8;
    public static final int THEME_BROWN = 9;
    public static final int THEME_GRAY = 10;
    public static final int THEME_BLUEGRAY = 11;

    public static int getThemeId(int theme){
        int themeId=0;
        switch (theme){
            case THEME_PRIMARY :{
                themeId = R.style.AppTheme;
                break;
            }
            case THEME_PINK  :
                themeId = R.style.AppTheme_PINK;
                break;
            case THEME_PURPLE  :
                themeId = R.style.AppTheme_PURPLE;
                break;
            case THEME_DEEPPURPLE  :
                themeId = R.style.AppTheme_DEEPPURPLE;
                break;
            case THEME_CYAN  :
                themeId = R.style.AppTheme_CYAN;
                break;
            case THEME_TEAL  :
                themeId = R.style.AppTheme_TEAL;
                break;
            case THEME_LIGHTGREEN  :
                themeId = R.style.AppTheme_LIGHTGREEN;
                break;
            case THEME_LIME  :
                themeId = R.style.AppTheme_LIME;
                break;
            case THEME_AMBER  :
                themeId = R.style.AppTheme_AMBER;
                break;
            case THEME_BROWN  :
                themeId = R.style.AppTheme_BROWN;
                break;
            case THEME_GRAY  :
                themeId = R.style.AppTheme_GRAY;
                break;
            case THEME_BLUEGRAY  :
                themeId = R.style.AppTheme_BLUEGRAY;
                break;
            default:
                break;
        }
        return themeId;
    }

    public static int getSlideThemeId(int theme){
        int themeId=0;
        switch (theme){
            case THEME_PRIMARY :{
                themeId = R.style.AppTheme_Slide;
                break;
            }
            case THEME_PINK  :
                themeId = R.style.AppTheme_PINK_Slide;
                break;
            case THEME_PURPLE  :
                themeId = R.style.AppTheme_PURPLE_Slide;
                break;
            case THEME_DEEPPURPLE  :
                themeId = R.style.AppTheme_DEEPPURPLE_Slide;
                break;
            case THEME_CYAN  :
                themeId = R.style.AppTheme_CYAN_Slide;
                break;
            case THEME_TEAL  :
                themeId = R.style.AppTheme_TEAL_Slide;
                break;
            case THEME_LIGHTGREEN  :
                themeId = R.style.AppTheme_LIGHTGREEN_Slide;
                break;
            case THEME_LIME  :
                themeId = R.style.AppTheme_LIME_Slide;
                break;
            case THEME_AMBER  :
                themeId = R.style.AppTheme_AMBER_Slide;
                break;
            case THEME_BROWN  :
                themeId = R.style.AppTheme_BROWN_Slide;
                break;
            case THEME_GRAY  :
                themeId = R.style.AppTheme_GRAY_Slide;
                break;
            case THEME_BLUEGRAY  :
                themeId = R.style.AppTheme_BLUEGRAY_Slide;
                break;
            default:
                break;
        }
        return themeId;
    }
    
    public static ArrayList<Theme> getThemeList(){
        ArrayList<Theme> themeArrayList = new ArrayList<>();
        themeArrayList.add(new Theme(0,R.color.colorPrimary, R.color.colorPrimaryDark, R.color.colorButton,"#0091EA"));
        themeArrayList.add(new Theme(1,R.color.primaryColorPink, R.color.primaryDarkColorPink, R.color.secondaryColorPink,"#448AFF"));
        themeArrayList.add(new Theme(2,R.color.primaryColorPurple, R.color.primaryDarkColorPurple, R.color.secondaryDarkColorPurple,"#b4c320"));
        themeArrayList.add(new Theme(3,R.color.primaryColorDeepPurple, R.color.primaryDarkColorDeepPurple, R.color.primaryColorPurple,"#9C27B0"));
        themeArrayList.add(new Theme(4,R.color.primaryColorCyan, R.color.primaryDarkColorCyan, R.color.primaryColorBrown,"#795548"));
        themeArrayList.add(new Theme(5,R.color.primaryColorTeal, R.color.primaryDarkColorTeal, R.color.secondaryDarkColorTeal,"#caa052"));
        themeArrayList.add(new Theme(6,R.color.primaryColorLightGreen, R.color.primaryDarkColorLightGreen, R.color.secondaryColorLightGreen,"#448AFF"));
        themeArrayList.add(new Theme(7,R.color.primaryColorLime, R.color.primaryDarkColorLime, R.color.primaryColorBlueGray,"#607D8B"));
        themeArrayList.add(new Theme(8,R.color.primaryColorAmber, R.color.primaryDarkColorAmber, R.color.secondaryColorAmber,"#03A9F4"));
        themeArrayList.add(new Theme(9,R.color.primaryColorBrown, R.color.primaryDarkColorBrown, R.color.md_teal_500,"#009688"));
        themeArrayList.add(new Theme(10,R.color.primaryColorGray, R.color.primaryDarkColorGray, R.color.secondaryColorGray,"#536DFE"));
        themeArrayList.add(new Theme(11,R.color.primaryColorBlueGray, R.color.primaryDarkColorBlueGray, R.color.secondaryDarkColorBlueGray,"#a6b320"));
        return themeArrayList;
    }

}

File: app/src/main/java/co/tpcreative/supersafe/common/util/ThemeUtil.kt
package co.tpcreative.supersafe.common.util
import co.tpcreative.supersafe.R
import co.tpcreative.supersafe.model.ThemeApp
import java.util.*
object ThemeUtil {
    const val THEME_PRIMARY = 0
    const val THEME_PINK = 1
    const val THEME_PURPLE = 2
    const val THEME_DEEPPURPLE = 3
    const val THEME_CYAN = 4
    const val THEME_TEAL = 5
    const val THEME_LIGHTGREEN = 6
    const val THEME_LIME = 7
    const val THEME_AMBER = 8
    const val THEME_BROWN = 9
    const val THEME_GRAY = 10
    const val THEME_BLUEGRAY = 11
    fun getThemeId(theme: Int): Int {
        var themeId = 0
        when (theme) {
            THEME_PRIMARY -> {
                themeId = R.style.AppTheme
            }
            THEME_PINK -> themeId = R.style.AppTheme_PINK
            THEME_PURPLE -> themeId = R.style.AppTheme_PURPLE
            THEME_DEEPPURPLE -> themeId = R.style.AppTheme_DEEPPURPLE
            THEME_CYAN -> themeId = R.style.AppTheme_CYAN
            THEME_TEAL -> themeId = R.style.AppTheme_TEAL
            THEME_LIGHTGREEN -> themeId = R.style.AppTheme_LIGHTGREEN
            THEME_LIME -> themeId = R.style.AppTheme_LIME
            THEME_AMBER -> themeId = R.style.AppTheme_AMBER
            THEME_BROWN -> themeId = R.style.AppTheme_BROWN
            THEME_GRAY -> themeId = R.style.AppTheme_GRAY
            THEME_BLUEGRAY -> themeId = R.style.AppTheme_BLUEGRAY
            else -> {
            }
        }
        return themeId
    }

    fun getSlideThemeId(theme: Int): Int {
        var themeId = 0
        when (theme) {
            ThemeUtil.THEME_PRIMARY -> {
                themeId = R.style.AppTheme
            }
            THEME_PINK -> themeId = R.style.AppTheme_PINK
            THEME_PURPLE -> themeId = R.style.AppTheme_PURPLE
            THEME_DEEPPURPLE -> themeId = R.style.AppTheme_DEEPPURPLE
            THEME_CYAN -> themeId = R.style.AppTheme_CYAN
            THEME_TEAL -> themeId = R.style.AppTheme_TEAL
            THEME_LIGHTGREEN -> themeId = R.style.AppTheme_LIGHTGREEN
            THEME_LIME -> themeId = R.style.AppTheme_LIME
            THEME_AMBER -> themeId = R.style.AppTheme_AMBER
            THEME_BROWN -> themeId = R.style.AppTheme_BROWN
            THEME_GRAY -> themeId = R.style.AppTheme_GRAY
            THEME_BLUEGRAY -> themeId = R.style.AppTheme_BLUEGRAY
            else -> {
            }
        }
        return themeId
    }

    fun getThemeList(): ArrayList<ThemeApp> {
        val themeAppArrayList: ArrayList<ThemeApp> = ArrayList<ThemeApp>()
        themeAppArrayList.add(ThemeApp(0, R.color.colorPrimary, R.color.colorPrimaryDark, R.color.colorAccent))
        themeAppArrayList.add(ThemeApp(1, R.color.primaryColorPink, R.color.primaryDarkColorPink, R.color.secondaryColorPink))
        themeAppArrayList.add(ThemeApp(2, R.color.primaryColorPurple, R.color.primaryDarkColorPurple, R.color.secondaryDarkColorPurple))
        themeAppArrayList.add(ThemeApp(3, R.color.primaryColorDeepPurple, R.color.primaryDarkColorDeepPurple, R.color.secondaryColorDeepPurple))
        themeAppArrayList.add(ThemeApp(4, R.color.primaryColorCyan, R.color.primaryDarkColorCyan, R.color.secondaryColorCyan))
        themeAppArrayList.add(ThemeApp(5, R.color.primaryColorTeal, R.color.primaryDarkColorTeal, R.color.secondaryDarkColorTeal))
        themeAppArrayList.add(ThemeApp(6, R.color.primaryColorLightGreen, R.color.primaryDarkColorLightGreen, R.color.secondaryColorLightGreen))
        themeAppArrayList.add(ThemeApp(7, R.color.primaryColorLime, R.color.primaryDarkColorLime, R.color.secondaryColorLime))
        themeAppArrayList.add(ThemeApp(8, R.color.primaryColorAmber, R.color.primaryDarkColorAmber, R.color.secondaryColorAmber))
        themeAppArrayList.add(ThemeApp(9, R.color.primaryColorBrown, R.color.primaryDarkColorBrown, R.color.secondaryColorBrown))
        themeAppArrayList.add(ThemeApp(10, R.color.primaryColorGray, R.color.primaryDarkColorGray, R.color.secondaryColorGray))
        themeAppArrayList.add(ThemeApp(11, R.color.primaryColorBlueGray, R.color.primaryDarkColorBlueGray, R.color.secondaryDarkColorBlueGray))
        return themeAppArrayList
    }
}